### PR TITLE
Add support for JSON `->` and `->>` operators

### PR DESCRIPTION
**Pull request by fulghum**
[MySQL `column->path` Documentation](https://dev.mysql.com/doc/refman/8.0/en/json-search-functions.html#operator_json-column-path)
[MySQL `column->>path` Documentation](https://dev.mysql.com/doc/refman/8.0/en/json-search-functions.html#operator_json-inline-path)

Fixes: https://github.com/dolthub/dolt/issues/5662

### CODE REVIEW

File: enginetest/queries/json_scripts.go [hunk 1]
```diff
@@ -333,6 +333,35 @@ var JsonScripts = []ScriptTest{
...
+				Expected: []sql.Row{{1, types.JSONDocument{Val: map[string]interface{}{"key1": 1, "key2": "\"abc\""}}}},
+			},
+			{
+				Query:    `select * from t where col1->>'$.key2' = 'abc';`,
+				Expected: []sql.Row{{1, types.JSONDocument{Val: map[string]interface{}{"key1": 1, "key2": "\"abc\""}}}},
+			},
+			{
+				Query:    `select * from t where col1->>'$.key2' = 'def';`,
+				Expected: []sql.Row{},
+			},
```
Comment (JCOR11599) [inline enginetest/queries/json_scripts.go:362]: Might be worth adding these test cases

Check if subqueries/aliasing works, and if operators can be nested
```sql
CREATE TABLE t (i INT PRIMARY KEY, j JSON);
INSERT INTO t VALUES (0, '{"a": "123", "outer": {"inner": 456}}');
SELECT k->"$.inner" from (SELECT j->"$.outer" AS k FROM t) sq;
+--------------+
| k->"$.inner" |
+--------------+
| 456          |
+--------------+
```

Check if array operations work
```sql
CREATE TABLE tj10 (a JSON, b INT);
INSERT INTO tj10 VALUES ('[3,10,5,"x",44]', 33), ('[3,10,5,17,[22,"y",66]]', 0);
SELECT a->"$[3]", a->"$[4][1]" FROM tj10;
+-----------+--------------+
| a->"$[3]" | a->"$[4][1]" |
+-----------+--------------+
| "x"       | NULL         |
| 17        | "y"          |
+-----------+--------------+

SELECT a->>"$[3]", a->>"$[4][1]" FROM tj10;
+------------+---------------+
| a->>"$[3]" | a->>"$[4][1]" |
+------------+---------------+
| x          | NULL          |
| 17         | y             |
+------------+---------------+
```

File: sql/planbuilder/scalar.go [hunk 2]
```diff
@@ -347,11 +347,18 @@ func (b *PlanBuilder) buildBinaryScalar(inScope *scope, be *ast.BinaryExpr) sql.
 		default:
 			return expression.NewArithmetic(l, r, be.Operator)
 		}
-	case
-		ast.JSONExtractOp,
-		ast.JSONUnquoteExtractOp:
-		err := sql.ErrUnsupportedFeature.New(fmt.Sprintf("(%s) JSON operators not supported", be.Operator))
-		b.handleErr(err)
+
+	case ast.JSONExtractOp, ast.JSONUnquoteExtractOp:
```
Comment (JCOR11599) [inline sql/planbuilder/scalar.go:351]: A plan/explain test to see that the operators are being converted properly could be nice.

File: sql/planbuilder/scalar.go [hunk 3]
```diff
@@ -347,11 +347,18 @@ func (b *PlanBuilder) buildBinaryScalar(inScope *scope, be *ast.BinaryExpr) sql.
...
+		jsonExtract, err := function.NewJSONExtract(l, r)
+		if err != nil {
+			b.handleErr(err)
+		}
+
+		if operator == ast.JSONUnquoteExtractOp {
+			return function.NewJSONUnquote(jsonExtract)
+		} else {
```
Comment (JCOR11599) [inline sql/planbuilder/scalar.go:359]: nit: `else` is not necessary

File: enginetest/queries/json_scripts.go [hunk 4]
```diff
@@ -333,6 +333,35 @@ var JsonScripts = []ScriptTest{
...
+				Query:    `select col1->'$.key1' from t;`,
+				Expected: []sql.Row{{types.JSONDocument{Val: 1}}},
+			},
+			{
+				Query:    `select col1->>'$.key2' from t;`,
+				Expected: []sql.Row{{"abc"}},
+			},
+			{
+				Query:    `select * from t where col1->'$.key1' = 1;`,
+				Expected: []sql.Row{{1, types.JSONDocument{Val: map[string]interface{}{"key1": 1, "key2": "\"abc\""}}}},
```
Comment (JCOR11599) [inline enginetest/queries/json_scripts.go:353]: Not sure which one is preferred, but we have a `MustJSON` method that is used in other enginetests

**Review by JCOR11599 — APPROVED** [review]
LGTM, small test case suggestions

File: enginetest/queries/json_scripts.go (hunk 1 above)
Comment (fulghum) [reply enginetest/queries/json_scripts.go:362]: Nice! 😄 Test coverage for JSON arrays was the missing piece I was hoping you'd find. Good job! 😄 

I like the subquery/aliasing suggestion, too. I'll add that in next.

File: enginetest/queries/json_scripts.go (hunk 4 above)
Comment (fulghum) [reply enginetest/queries/json_scripts.go:353]: Ooh... that is a super handy util function. Thanks for pointing it out!

**Pull request merged**